File: src/test/groovy/graphql/ScalarsQuerySchema.java
package graphql;


import graphql.schema.*;

import static graphql.schema.GraphQLArgument.newArgument;
import static graphql.schema.GraphQLFieldDefinition.newFieldDefinition;
import static graphql.schema.GraphQLObjectType.newObject;

import java.math.BigDecimal;
import java.math.BigInteger;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.CompletionStage;

public class ScalarsQuerySchema {

    public static final DataFetcher inputDF = new DataFetcher() {
        @Override
        public CompletionStage<Object> fetch(DataFetchingEnvironment environment) {
            return CompletableFuture.completedFuture(environment.argument("input"));
        }
    };

    public static final GraphQLObjectType queryType = Companion.newObject()
                                                               .name("QueryType")
                                                               /** Static Scalars */
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("bigInteger")
                                                                               .type(Scalars.INSTANCE.getGraphQLBigInteger())
                                                                               .staticValue(BigInteger.valueOf(9999)))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("bigDecimal")
                                                                               .type(Scalars.INSTANCE.getGraphQLBigDecimal())
                                                                               .staticValue(BigDecimal.valueOf(1234.0)))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("floatNaN")
                                                                               .type(Scalars.INSTANCE.getGraphQLFloat())
                                                                               .staticValue(Double.NaN))




                                                               /** Scalars with input of same type, value echoed back */
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("bigIntegerInput")
                                                                               .type(Scalars.INSTANCE.getGraphQLBigInteger())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(new GraphQLNonNull(Scalars.INSTANCE.getGraphQLBigInteger())))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("bigDecimalInput")
                                                                               .type(Scalars.INSTANCE.getGraphQLBigDecimal())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(new GraphQLNonNull(Scalars.INSTANCE.getGraphQLBigDecimal())))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("floatNaNInput")
                                                                               .type(Scalars.INSTANCE.getGraphQLFloat())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(new GraphQLNonNull(Scalars.INSTANCE.getGraphQLFloat())))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("stringInput")
                                                                               .type(Scalars.INSTANCE.getGraphQLString())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(new GraphQLNonNull(Scalars.INSTANCE.getGraphQLString())))
                                                                               .dataFetcher(inputDF))






                                                               /** Scalars with input of String, cast to scalar */
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("bigIntegerString")
                                                                               .type(Scalars.INSTANCE.getGraphQLBigInteger())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("bigDecimalString")
                                                                               .type(Scalars.INSTANCE.getGraphQLBigDecimal())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("floatString")
                                                                               .type(Scalars.INSTANCE.getGraphQLFloat())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("longString")
                                                                               .type(Scalars.INSTANCE.getGraphQLLong())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("intString")
                                                                               .type(Scalars.INSTANCE.getGraphQLInt())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("shortString")
                                                                               .type(Scalars.INSTANCE.getGraphQLShort())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .field(Companion.newFieldDefinition()
                                                                               .name("byteString")
                                                                               .type(Scalars.INSTANCE.getGraphQLByte())
                                                                               .argument(Companion.newArgument()
                                       .name("input")
                                       .type(Scalars.INSTANCE.getGraphQLString()))
                                                                               .dataFetcher(inputDF))
                                                               .build();


    public static final GraphQLSchema scalarsQuerySchema = GraphQLSchema.Companion.newSchema()
                                                                                  .query(queryType)
                                                                                  .build();
}

File: src/test/groovy/graphql/ScalarsQuerySchema.kt
package graphql


import graphql.schema.*

import java.math.BigDecimal
import java.math.BigInteger
import java.util.concurrent.CompletableFuture.completedFuture

private fun <T> inputFetcher(): DataFetcher<T> = {
    completedFuture(it.argument("input"))
}


private val scalarQueryType = newObject {
    name = "QueryType"

    /** Static Scalars  */
    field<BigInteger> {
        name = "bigInteger"
        staticValue = BigInteger.valueOf(9999)
    }
    field<BigDecimal> {
        name = "bigDecimal"
        staticValue = BigDecimal.valueOf(1234.0)
    }
    field<Double> {
        name = "floatNaN"
        staticValue = Double.NaN
    }

    /** Scalars with input of same type, value echoed back  */
    field<BigInteger> {
        name = "bigIntegerInput"
        argument {
            name = "input"
            type = GraphQLNonNull(GraphQLBigInteger)
        }
        fetcher = inputFetcher()
    }
    field<BigDecimal> {
        name = "bigDecimalInput"
        argument {
            name = "input"
            type = GraphQLNonNull(GraphQLBigDecimal)
        }
        fetcher = inputFetcher()
    }
    field<Double> {
        name = "floatNaNInput"
        argument {
            name = "input"
            type = GraphQLNonNull(GraphQLFloat)
        }
        fetcher = inputFetcher()
    }
    field<String> {
        name = "stringInput"
        argument {
            name = "input"
            type = GraphQLStringNonNull
        }
        fetcher = inputFetcher()
    }

    /** Scalars with input of String, cast to scalar  */
    field<BigInteger> {
        name = "bigIntegerString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
    field<BigDecimal> {
        name = "bigDecimalString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
    field<Double> {
        name = "floatString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
    field<Long> {
        name = "longString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
    field<Int> {
        name = "intString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
    field<Short> {
        name = "shortString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
    field<Byte> {
        name = "byteString"
        argument {
            name = "input"
            type = GraphQLString
        }
        fetcher = inputFetcher()
    }
}

val scalarsQuerySchema = newSchema {
    query = scalarQueryType
}
